File: src/commands/uwu.js
const { SlashCommandBuilder } = require("discord.js");
const Uwuifier = require("uwuifier");

let uwuifier = new Uwuifier.default({
  spaces: {
    faces: 0.125,
    actions: 0,
    stutters: 0.125,
  },
  words: 1,
  exclamations: 1,
});

async function execute(interaction) {
  let text = interaction.options.getString("text");

  text = text.replace("@everyone", "at everyone");

  let uwu = uwuifier.uwuifySentence(text);

  if (uwu.length > 2000) {
    interaction.reply("Your message was too long.");
  }

  let name = uwuifier.uwuifyWords(interaction.user.username);
  let avatarUrl = interaction.user.avatarURL();

  let webhooks = await interaction.channel.fetchWebhooks().catch(console.error);
  if (webhooks.size == 0) {
    await interaction.channel
      .createWebhook({
        name: "UwU webhook",
        avatar:
          "https://media.discordapp.net/attachments/1015273149115416596/1025011813026373682/licc.png",
      })
      .then(console.log)
      .catch(console.error);
    webhooks = await interaction.channel.fetchWebhooks().catch(console.error); // :skull:
  }

  let webhook = webhooks.first();

  webhook.send({
    content: uwu,
    username: name,
    avatarURL: avatarUrl,
  });

  await interaction.reply({ content: "ok.", ephemeral: true });
}

module.exports = {
  data: new SlashCommandBuilder()
    .setName("uwu")
    .setDescription("UwUifies your text.")
    .addStringOption(option =>
      option
        .setName("text")
        .setDescription("The text to uwuify")
        .setRequired(true)
    ),
  execute,
};

File: src/commands/uwu.rs
use crate::{Context, Error};
use poise::{
    command,
    serenity_prelude::{
        CreateAttachment, CreateWebhook, ExecuteWebhook, Webhook,
    },
    CreateReply,
};

#[command(slash_command, prefix_command, guild_only)]
/// uwuify a string
pub async fn uwu(ctx: Context<'_>, #[rest] text: String) -> Result<(), Error> {
    let out = crate::uwu::uwuify(text);
    if ctx.prefix() == "/" {
        let m = CreateReply::default().content(";").ephemeral(true);
        ctx.send(m).await?;
    }

    let hook = match get_webhook(ctx).await {
        Ok(hook) => hook,
        Err(e) => {
            let content = format!("Unable to find/create webhook: {}", e);
            let message = CreateReply::default().content(content).ephemeral(true);
            ctx.send(message).await?;
            return Ok(());
        }
    };

    let user = match ctx.author_member().await {
        Some(u) => u,
        None => unreachable!(),
    };

    let m = ExecuteWebhook::new()
        .username(user.display_name())
        .avatar_url(user.face())
        .content(out);

    hook.execute(ctx, false, m).await?;

    Ok(())
}

async fn create_webhook(ctx: Context<'_>) -> Result<Webhook, Error> {
    let channel = ctx.channel_id();

    let mut create_hook = CreateWebhook::new("AutoVirt: uwu");

    if let Some(pfp) = &ctx.data().bot_pfp {
        let attachment = CreateAttachment::url(ctx, pfp).await?;
        create_hook = create_hook.avatar(&attachment);
    }

    let hook = channel.create_webhook(ctx, create_hook).await?;

    Ok(hook)
}

async fn get_webhook(ctx: Context<'_>) -> Result<Webhook, Error> {
    let hooks = ctx.channel_id().webhooks(&ctx).await?;

    let hook = match hooks.iter().find(|w| w.token.is_some()) {
        Some(w) => Ok(w.to_owned()),
        None => create_webhook(ctx).await,
    };

    hook
}
